app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java

```diff
@@ -604,8 +610,13 @@ public void afterTextChanged(final Editable s) {
                     s.removeSpan(span);
                 }
 
-                final String newText = searchEditText.getText().toString();
+                final String newText = searchEditText.getText().toString().trim();
                 suggestionPublisher.onNext(newText);
+
+                if (isPastedText) {
+                    // trim pasted text
+                    searchEditText.setText(newText);
```

Comment:
Why trim it in the search field, too? Let the user type or paste what he wants and:
- trim the string before passing it to `suggestionPublisher`
- when the user presses search, set the trimmed string in the search bar, search for the trimmed string and save the trimmed string in the db

Reply:
The above two behaviors have been implemented.

As trimming when pasting, it is inconsistent with the behavior of other search bars. I thought it's often unintentional for users to paste text with leading and trailing spaces. So I made this additional change.